app/ports/models/port.py

```diff
@@ -283,6 +283,24 @@ def mark_deleted(cls, dict_of_portdirs_with_ports):
                     port.active = False
                     port.save()
 
+    # Run through all the ports to identify deleted ports
+    @classmethod
+    def full_deleted_ports_run(cls):
+        all_ports_from_json = set()
+        data = Port.PortIndexUpdateHandler().sync_and_open_file()
```

Comment:
I need to check again, but does running this function always fetch the latest version of the json file?

I kind of feel that the functionality should be separate from rsync-ing the file.

Reply:
Yes, the function `sync_and_open_file` first fetches the latest file and then reads it.